src/OpenLoco/Audio/Audio.cpp

```diff
@@ -875,13 +878,117 @@ namespace OpenLoco::Audio
+        int32_t maxVolume = -3500;
+
+        if (!Game::hasFlags((1u << 0)) && mainViewport != nullptr)
+        {
+            maxVolume = getAmbientMaxVolume(mainViewport->zoom);
+            const auto centre = mainViewport->getCentreMapPosition();
+            const auto topLeft = Map::TilePos2{ centre } - Map::TilePos2{ 5, 5 };
+            const auto bottomRight = topLeft + Map::TilePos2{ 11, 11 };
+            Map::TilePosRangeView searchRange(topLeft, bottomRight);
+            size_t waterCount = 0;      // bl
+            size_t wildernessCount = 0; // bh
+            size_t treeCount = 0;       //cx
+            for (auto& tilePos : searchRange)
+            {
+                const auto tile = Map::TileManager::get(tilePos);
+                bool passedSurface = false;
+                for (const auto& el : tile)
+                {
+                    auto* elSurface = el.as<Map::SurfaceElement>();
+                    if (elSurface != nullptr)
+                    {
+                        passedSurface = true;
+                        if (elSurface->water() != 0)
+                        {
+                            waterCount++;
+                            break;
+                        }
+                        else if (elSurface->var_4_E0() && elSurface->isLast())
+                        {
+                            wildernessCount++;
+                            break;
+                        }
+                        else if (elSurface->baseZ() >= 64 && elSurface->isLast())
+                        {
+                            wildernessCount++;
+                            break;
+                        }
+                        continue;
+                    }
+                    auto* elTree = el.as<Map::TreeElement>();
+                    if (passedSurface && elTree != nullptr)
+                    {
+                        const auto* treeObj = ObjectManager::get<TreeObject>(elTree->treeObjectId());
+                        if (!(treeObj->flags & TreeObjectFlags::droughtResistant))
+                        {
+                            treeCount++;
+                        }
+                    }
+                }
+            }
+
+            if (waterCount > 60)
+            {
+                ambientSound = PathId::css3;
+            }
+            else if (wildernessCount > 60)
+            {
+                ambientSound = PathId::css2;
+            }
+            else if (treeCount > 30)
+            {
+                ambientSound = PathId::css4;
+            }
+        }
+        auto* channel = getChannel(ChannelId::ambient);
+        if (channel == nullptr)
+        {
+            return;
+        }
+        // In these situations quieten until channel stopped
+        if (!ambientSound.has_value() || (channel->isPlaying() && _chosenAmbientNoisePathId != *ambientSound))
+        {
+            const auto newVolume = channel->getAttributes().volume - 100;
+            if (newVolume < -3500)
```

Comment:
I think all these numbers should be constants, especially the ones that are used multiple times like `-3500`.

Reply:
This is actually the minimum volume. Its saying if we are below the min then we can stop the sound and free the channel for a new sound.

Reply:
oh right I see. in that case we should make a `constexpr int32_t minVolume;` then, that would make all of this much clearer to the reader what the value means